### PR TITLE
DOC: update ndarray.item docstring

### LINKED ISSUE #14139: DOC: np.array(<numpy data type>, dtype=object).item() doesn't return python scalar
### Reproducing code example:

```python
Python 3.7.3 (default, Mar 27 2019, 16:54:48)
[Clang 4.0.1 (tags/RELEASE_401/final)] :: Anaconda, Inc. on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import numpy as np; np.__version__
'1.17.0'
>>> type(np.array([np.bool_(True)], dtype=object).item())
<class 'numpy.bool_'>
>>> type(np.array([np.int64(1)], dtype=object).item())
<class 'numpy.int64'>
```

From the `item` [docstring](https://docs.scipy.org/doc/numpy/reference/generated/numpy.ndarray.item.html), I would expect the above example to still return `bool` and `int` respectively even if `dtype=object`. If this is the expected behavior, it would be great if the documentation was updated with this edge case.

### PR DESCRIPTION
**Pull request by liang3zy22**
[skip actions][skip cirrus][skip azp]

Closes #14139.

### CODE REVIEW

File: numpy/_core/_add_newdocs.py [hunk 1]
```diff
@@ -3619,6 +3619,10 @@
     access to elements of the array and doing arithmetic on elements of the
     array using Python's optimized math.
 
+    There are some edge cases when numpy scalar is returned. in this scenario,
+    `item` can be recursively called to get the standard python scalar. But for
+    `numpy.longdouble` type, `item` would return itself.
+
```
Comment (rossbar) [inline numpy/_core/_add_newdocs.py:3625]: IMO this feels like information overload and brings too much attention to a specific corner case. If anything, I'd vote for [@seberg's wording suggestion](https://github.com/numpy/numpy/issues/14139#issuecomment-515792885): 

> for an array with object dtype the elements are returned as-is

**Review by rossbar — COMMENTED** [review]
I think the confusion in #14139 arises from the first sentence in the docstring:

> Copy an element of an array to a standard Python scalar and return it.

I think the best way to "fix" this issue would be to modify the docstring like so:

```
a.item(*args)

Copy an element of an array to a standard Python scalar and return it.

For an array with object dtype, elements are returned as-is.
```

File: numpy/_core/_add_newdocs.py (hunk 1 above)
Comment (liang3zy22) [reply numpy/_core/_add_newdocs.py:3625]: I think it be good. But I have added some example codes, these information is to describe these examples.

**Comment by liang3zy22** [comment]
@rossbar , as you suggested, I used seberg's word suggestion. And using comment to describe the example codes.

File: numpy/_core/_add_newdocs.py [hunk 2]
```diff
@@ -3636,6 +3636,19 @@
...
+    >>> a.item() #return np.int64
+    np.int64(1)
+    >>> a.item().item() #return python scalar
+    1
+    >>> a = np.array([np.longdouble(1.11)], dtype=object)
+    >>> a.item() #return np.longdouble
+    np.longdouble('1.1111111')
+    >>> a.item().item() #still return np.longdouble
+    np.longdouble('1.1111111')
+
```
Comment (rossbar) [inline numpy/_core/_add_newdocs.py:3651]: Let's remove these parts of the example - my feeling that this is too in-the-weeds for a docstring example and is generally not of interest to typical users (the folks from the relevant discussion in the original issue are downstream library developers).

Comment (liang3zy22) [reply numpy/_core/_add_newdocs.py]: Done.

**Review by rossbar — APPROVED** [review]
Thanks @liang3zy22 !

**Pull request merged**